airflow/operators/dagrun_operator.py

```diff
@@ -126,3 +144,22 @@ def execute(self, context: Dict):
                 dag.clear(start_date=self.execution_date, end_date=self.execution_date)
             else:
                 raise e
+
+        if self.wait_for_completion:
+            # wait for dag to complete
+            while True:
+                self.log.info(
+                    'Waiting for %s on %s to become allowed state %s ...',
+                    self.trigger_dag_id,
+                    dag_run.execution_date,
+                    self.allowed_states,
+                )
+                dag_run.refresh_from_db()
+                state = dag_run.state
+                if state in self.failed_states:
+                    raise AirflowException(f"{self.trigger_dag_id} failed with failed states {state}")
```

Comment:
@kukigai the same logic can be used in ExternalTaskSensor - would you like to refactor this sensor in separate PR?

Reply:
@turbaszek 

Yes, I was thinking about that too.
I will send separate PR once this is done.

Reply:
@kukigai have you been able to use the same logic in ExternalTaskSensor?

Reply:
@turbaszek 

Actually, we can not apply the same logic as here.
The reason is
1. external task sensor doesn't have dag_run object so need to look up database instead of using dag_run object which is typically created by dag run operator.
2. sensor has poke interval concept already so not need to do while loop like this.
3. https://github.com/apache/airflow/blob/master/airflow/sensors/external_task_sensor.py#L184-L190 already handles the failed task handling.

Let me know your opinion.